Mathlib/Geometry/Manifold/IntegralCurve.lean

```diff
@@ -0,0 +1,153 @@
+
+/-- An interior point of a manifold is a point whose image in the model vector space is in the
+interior of the chart's target. -/
+def ModelWithCorners.IsInteriorPoint (x : M) := extChartAt I x x ∈ interior (extChartAt I x).target
+
+lemma ModelWithCorners.Boundaryless.isOpen_target
+    [I.Boundaryless] {x : M} : IsOpen (extChartAt I x).target := by
+rw [extChartAt_target, ModelWithCorners.Boundaryless.range_eq_univ, Set.inter_univ]
+exact (ModelWithCorners.continuous_symm _).isOpen_preimage _ (LocalHomeomorph.open_target _)
+
+lemma ModelWithCorners.Boundaryless.isInteriorPoint
+    [I.Boundaryless] {x : M} : I.IsInteriorPoint x := by
+rw [ModelWithCorners.IsInteriorPoint,
+  IsOpen.interior_eq ModelWithCorners.Boundaryless.isOpen_target]
+exact LocalEquiv.map_source _ (mem_extChartAt_source _ _)
+
+end
+
+variable
+  {E : Type*} [NormedAddCommGroup E] [NormedSpace ℝ E] [CompleteSpace E]
+  {H : Type*} [TopologicalSpace H] {I : ModelWithCorners ℝ E H}
+  {M : Type*} [TopologicalSpace M] [ChartedSpace H M] [SmoothManifoldWithCorners I M]
+  {E' : Type*} [NormedAddCommGroup E'] [NormedSpace ℝ E']
+  {H' : Type*} [TopologicalSpace H'] {I' : ModelWithCorners ℝ E' H'}
+  {M' : Type*} [TopologicalSpace M'] [ChartedSpace H' M'] [SmoothManifoldWithCorners I' M']
+  {v : (x : M) → TangentSpace I x} {x₀ : M}
+  (hv : ContMDiffAt I I.tangent 1 (fun x => (⟨x, v x⟩ : TangentBundle I M)) x₀) (t₀ : ℝ)
+
+def HasIntegralCurveAt (v : (x : M) → TangentSpace I x) (γ : ℝ → M) (x₀ : M) (t₀ : ℝ) :=
+  γ t₀ = x₀ ∧ ∃ ε > (0 : ℝ), ∀ (t : ℝ), t ∈ Set.Ioo (t₀ - ε) (t₀ + ε) →
+    HasMFDerivAt 𝓘(ℝ, ℝ) I γ t ((1 : ℝ →L[ℝ] ℝ).smulRight (v (γ t)))
+
+/-- For any continuously differentiable vector field and any chosen non-boundary point `x₀` on the
+  manifold, an integral curve `γ : ℝ → M` exists such that `γ t₀ = x₀` and the tangent vector of `γ`
+  at `t` coincides with the vector field at `γ t` for all `t` within an open interval around `t₀`.-/
+theorem exists_integralCurve_of_contMDiff_tangent_section (hx : I.IsInteriorPoint x₀) :
+    ∃ (γ : ℝ → M), HasIntegralCurveAt v γ x₀ t₀ := by
+rw [contMDiffAt_iff] at hv
+obtain ⟨_, hv⟩ := hv
+have hI : Set.range I ∈ nhds (extChartAt I x₀ x₀)
+· rw [mem_nhds_iff]
+  refine ⟨interior (extChartAt I x₀).target,
+    subset_trans interior_subset (extChartAt_target_subset_range ..),
+    isOpen_interior, hx⟩
+obtain ⟨f, hf1, ε1, hε1, hf2⟩ :=
+  exists_forall_hasDerivAt_Ioo_eq_of_contDiffAt t₀ (ContDiffAt.snd (hv.contDiffAt hI))
+rw [←Real.ball_eq_Ioo] at hf2
+-- use continuity of f to extract ε2 so that for t ∈ Real.ball t₀ ε2,
+-- f t ∈ interior (extChartAt I x₀).target
+have hcont := (hf2 t₀ (Real.ball_eq_Ioo .. ▸ Metric.mem_ball_self hε1)).continuousAt
+rw [continuousAt_def, hf1] at hcont
+have hnhds : f ⁻¹' (interior (extChartAt I x₀).target) ∈ nhds t₀ :=
+  hcont _ (IsOpen.mem_nhds isOpen_interior hx)
+rw [Metric.mem_nhds_iff] at hnhds
+obtain ⟨ε2, hε2, hf3⟩ := hnhds
+simp_rw [Set.subset_def, Set.mem_preimage] at hf3
```

Comment:
Given all the Set lemmas you're using, `open`ing Set might be a nice minor clean-up.